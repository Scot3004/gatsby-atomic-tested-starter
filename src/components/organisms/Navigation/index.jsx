import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'gatsby';
import Card from '../../atoms/Card';
import CardContent from '../../atoms/CardContent';
import FeatherIcon from '../../molecules/FeatherIcon';

export const Navigation = ({ menuLinks }) => (
  <Card>
    <CardContent>
      {menuLinks.map(link => (
        <Link to={link.link} key={link.name}>
          <FeatherIcon name={link.icon} />
          {link.name}
        </Link>
      ))}
    </CardContent>
  </Card>
);

Navigation.propTypes = {
  menuLinks: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string,
      link: PropTypes.string
    })
  ).isRequired
};

export default Navigation;
