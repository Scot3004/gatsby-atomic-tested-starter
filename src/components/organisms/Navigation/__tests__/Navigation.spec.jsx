import React from 'react';
import 'jest-styled-components';
import ShallowRenderer from 'react-test-renderer/shallow';
import Navigation from '..';

const menuLinks = [
  {
    icon: 'calendar',
    name: 'name stub',
    link: 'link_stub'
  }
];

describe('When I want to navigate through the pages', () => {
  const renderer = new ShallowRenderer();
  renderer.render(<Navigation menuLinks={menuLinks} />);
  test('Then i could see a navigation bar', () => {
    const tree = renderer.getRenderOutput();
    expect(tree).toMatchSnapshot();
  });
});
