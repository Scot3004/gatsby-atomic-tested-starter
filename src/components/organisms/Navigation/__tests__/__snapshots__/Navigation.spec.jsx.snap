// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When I want to navigate through the pages Then i could see a navigation bar 1`] = `
<ForwardRef>
  <ForwardRef
    activeClassName="active"
    to="link_stub"
  >
    <FeatherIcon
      name="calendar"
      size={24}
    />
    name stub
  </ForwardRef>
</ForwardRef>
`;
